src/restLink.ts

```diff
@@ -1393,12 +1422,28 @@ export class RestLink extends ApolloLink {
             resolverOptions,
           )
             .then(data => {
-              setContext({
-                restResponses: (context.restResponses || []).concat(
+              /** Composability Modification:
+               * not sure why the code is caching setContext but we want to make sure
+               * we're setting context the the right point in the chain and using a cached
+               * setContext can break this
+               */
+              operation.setContext({
+                /** Composability Modification:
+                 * get the context fresh since it could have changed between when it was cached above
+                 * and when this code is called
+                 * we use this data field to preserve the data from previous rest link requests
+                 * aside: what is restResponses used for? I can't find refs to it
+                 */
```

Comment:
```suggestion
                /**
                 * get the context fresh since it could have changed between when it was cached above
                 */
```

No idea what `restResponses` is used for, could be a leftover from an older generation of apollo-link chaining, or maybe a changed approach for implementing some feature.

I took over maintenance of this link years ago, but hopped into the project after a proof-of-concept already existed.

Reply:
Should I remove it? That wouldn't be a backwards compatible change if someone were using it though.

Reply:
I’m fine ignoring it for now!